controller_manager:
  ros__parameters:
    update_rate: 30  # Zmniejszona częstotliwość aktualizacji dla symulacji
    controller_manager_timeout: 10.0  # Znacznie zwiększony timeout dla symulacji

    joint_state_broadcaster:
      type: joint_state_broadcaster/JointStateBroadcaster

    leg1_controller:
      type: position_controllers/JointGroupPositionController

    leg2_controller:
      type: position_controllers/JointGroupPositionController

    leg3_controller:
      type: position_controllers/JointGroupPositionController
    
    leg4_controller:
      type: position_controllers/JointGroupPositionController
    
    leg5_controller:
      type: position_controllers/JointGroupPositionController
    
    leg6_controller:
      type: position_controllers/JointGroupPositionController

joint_state_broadcaster:
  ros__parameters:
    publish_rate: 30  # Zmniejszona częstotliwość publikacji dla symulacji

leg1_controller:
  ros__parameters:
    joints:
      - joint1_1
      - joint2_1
      - joint3_1
    interface_name: position
    initial_position:
      joint1_1: 0.1
      joint2_1: 0.5
      joint3_1: 0.6

leg2_controller:
  ros__parameters:
    joints:
      - joint1_2
      - joint2_2
      - joint3_2
    interface_name: position
    initial_position:
      joint1_2: 0.0
      joint2_2: 0.5
      joint3_2: 0.6

leg3_controller:
  ros__parameters:
    joints:
      - joint1_3
      - joint2_3
      - joint3_3
    interface_name: position
    initial_position:
      joint1_3: -0.1
      joint2_3: 0.5
      joint3_3: 0.6

leg4_controller:
  ros__parameters:
    joints:
      - joint1_4
      - joint2_4
      - joint3_4
    interface_name: position
    initial_position:
      joint1_4: 0.1
      joint2_4: 0.5
      joint3_4: 0.6

leg5_controller:
  ros__parameters:
    joints:
      - joint1_5
      - joint2_5
      - joint3_5
    interface_name: position
    initial_position:
      joint1_5: 0.0
      joint2_5: 0.5
      joint3_5: 0.6

leg6_controller:
  ros__parameters:
    joints:
      - joint1_6
      - joint2_6
      - joint3_6
    interface_name: position
    initial_position:
      joint1_6: -0.1
      joint2_6: 0.5
      joint3_6: 0.6